src/lib.rs

```diff
@@ -223,9 +223,44 @@ fn components(blurhash: &str) -> Result<(usize, usize), Error> {
     Ok((num_x, num_y))
 }
 
+#[cfg(feature = "image")]
+/// Calculates the blurhash for an [DynamicImage][image::DynamicImage] using the given x and y component counts.
+pub fn encode_image(
```

Comment:
I think we usually put the feature gates after the doc comment.

```suggestion
/// Calculates the blurhash for an [DynamicImage][image::DynamicImage] using the given x and y component counts.
#[cfg(feature = "image")]
pub fn encode_image(
```

Additionally, this method reallocates the image (`to_rgba8()` is always allocating, `into_rgba8()` is not). We might just want to accept the rgba8 version, and leave the conversion to the caller. Or we provide both methods: `encode_dynamic_image()` and `encode_image`.

Reply:
I decided to do `RgbaImage` only, conversion is simple enough on the user-side.